src/Bicep.Cli/Services/TemplateEvaluation.cs

```diff
@@ -18,14 +18,26 @@
 
 namespace Bicep.Cli.Services
 {
-    public class TemplateEvaluatorService
+    public class TemplateEvaluation
     {
         private const string DummyTenantId = "";
         private const string DummyManagementGroupName = "";
         private const string DummySubscriptionId = "";
         private const string DummyResourceGroupName = "";
         private const string DummyLocation = "";
         private static readonly Regex templateSchemaPattern = new Regex(@"https?://schema\.management\.azure\.com/schemas/[0-9a-zA-Z-]+/(?<templateType>[a-zA-Z]+)Template\.json#?", RegexOptions.Compiled);
+        private InvalidOperationException? error;
+        private Template? resultTemplate;
+        private Dictionary<string, bool>? assertions;
+        private bool isSkipped;
+        private bool isSuccessful;
+
+        public InvalidOperationException? Error => error;
+        public Template? ResultTemplate => resultTemplate;
+        public Dictionary<string, bool> Assertions => assertions ?? new Dictionary<string, bool>();
+        public Dictionary<string, bool> FailedAssertions => assertions?.Where(a => !a.Value).ToDictionary(a => a.Key, a => a.Value) ?? new Dictionary<string, bool>();
+        public bool Success => isSuccessful;
+        public bool Skip => isSkipped;
```

Comment:
We've being trying to avoid making things mutable in Bicep. I think it would be better to create dedicated immutable `EvaluationResult` class to encapsulate the validation result properties.